pandas/core/nanops.py

```diff
@@ -1417,3 +1417,170 @@ def nanpercentile(values, q, axis, na_value, mask, ndim, interpolation):
             return result
     else:
         return np.percentile(values, q, axis=axis, interpolation=interpolation)
+
+
+def kleene_or(
+    left: Union[bool, np.ndarray],
+    right: Union[bool, np.ndarray],
+    left_mask: Optional[np.ndarray],
+    right_mask: Optional[np.ndarray],
+):
+    """
+    Boolean ``or`` using Kleene logic.
+
+    Values are NA where we have ``NA | NA`` or ``NA | False``.
+    ``NA | True`` is considered True.
+
+    Parameters
+    ----------
+    left, right : ndarray, NA, or bool
+        The values of the array.
+    left_mask, right_mask : ndarray, optional
+        The masks. When
+
+    Returns
+    -------
+    result, mask: ndarray[bool]
+        The result of the logical or, and the new mask.
+    """
+    # To reduce the number of cases, we ensure that `left` & `left_mask`
+    # always come from an array, not a scalar. This is safe, since because
+    # A | B == B | A
+    if left_mask is None:
+        return kleene_or(right, left, right_mask, left_mask)
+
+    assert isinstance(left, np.ndarray)
+
+    raise_for_nan(right, method="or")
+
+    if right is libmissing.NA:
```

Comment:
should libmissing.NA be in the signature for left or right?

Reply:
Added libmissing.NAType.